tests/integration/dbapi/async/test_queries_async.py

```diff
@@ -341,6 +409,77 @@ async def test_set_invalid_parameter(connection: Connection):
     with connection.cursor() as c:
         assert len(c._set_parameters) == 0
         with raises(OperationalError):
-            await c.execute("set some_invalid_parameter = 1")
+            await c.execute("SET some_invalid_parameter = 1")
 
         assert len(c._set_parameters) == 0
+
+
+async def test_server_side_async_execution_query(connection: Connection) -> None:
+    """Make an sql query and receive an id back."""
+    with connection.cursor() as c:
+        query_id = await c.execute("SELECT 1", [], async_execution=True)
+    assert (
+        type(query_id) is str and query_id
+    ), "Invalid query id was returned from server-side async query."
+
+
+async def test_server_side_async_execution_cancel(connection: Connection) -> None:
+    """Test cancel."""
+    with connection.cursor() as c:
+        try:
+            await c.execute(
+                "CREATE DIMENSION TABLE IF NOT EXISTS test_tbl (id int, name string)"
+            )
+            query_id = await c.execute(
+                LONG_INSERT,
+                async_execution=True,
+            )
+            # Cancel, then check that status is cancelled.
+            await c.cancel(query_id)
+            await status_loop(
+                query_id,
+                "cancel",
+                c,
+                final_status=QueryStatus.CANCELED_EXECUTION,
+            )
+        finally:
+            await c.execute("DROP TABLE IF EXISTS test_tbl")
+
+
+async def test_server_side_async_execution_get_status(connection: Connection) -> None:
+    """
+    Test get_status(). Test for three ending conditions: PARSE_ERROR,
+    STARTED_EXECUTION, ENDED_EXECUTION.
+    """
+    with connection.cursor() as c:
+        try:
+            await c.execute(
+                "CREATE DIMENSION TABLE IF NOT EXISTS test_tbl (id int, name string)"
```

Comment:
This can also be a global. In fact, since this as well as `LONG_INSERT` are reused in both sync and async tests they can be placed at the dbapi/ conftest level.

Reply:
I made this and the `DROP TABLE` below both globals, but I don't know how to move them out of their modules and up a level: I couldn't get the imports to work, and  pytest won't import anything other than fixtures from the conftest files, as far as I can tell.

Reply:
Actually, I had to reopen this. Can you tell me how to get the import to work? It seems to be beyond my Python abilities.